embulk-input-s3/src/main/java/org/embulk/input/s3/S3FileInputPlugin.java

```diff
@@ -21,31 +18,38 @@
         @Config("endpoint")
         @ConfigDefault("null")
         public Optional<String> getEndpoint();
+
+        @Config("region")
+        @ConfigDefault("null")
+        public Optional<String> getRegion();
     }
 
+    private static final Logger log = Exec.getLogger(S3FileInputPlugin.class);
+
     @Override
-    protected Class<? extends PluginTask> getTaskClass()
-    {
+    protected Class<? extends PluginTask> getTaskClass() {
         return S3PluginTask.class;
     }
 
     @Override
     protected AmazonS3 newS3Client(PluginTask task)
     {
         S3PluginTask t = (S3PluginTask) task;
+        Optional<String> endpoint = t.getEndpoint();
+        Optional<String> region = t.getRegion();
 
         AmazonS3ClientBuilder builder = super.defaultS3ClientBuilder(t);
 
-        if (t.getEndpoint().isPresent()) {
-            String endpoint = t.getEndpoint().get();
-            builder.setEndpointConfiguration(new EndpointConfiguration(
-                    endpoint,
-                    // Although client will treat endpoint's region as the signer region
-                    // if we left this as null, but such that behaviour is undocumented,
-                    // so it is explicitly calculated here for future-proofing.
-                    parseRegion(
-                            toUri(endpoint, getClientConfiguration(task)).getHost(),
-                            S3_SERVICE_NAME)));
+        if (endpoint.isPresent()) {
+            if (region.isPresent()) {
+                log.warn("Either configure endpoint or region, " +
+                        "if both is specified only the endpoint will be in effect.");
+            }
+            builder.setEndpointConfiguration(new EndpointConfiguration(endpoint.get(), null));
+        } else if (region.isPresent()) {
+            builder.setRegion(region.get());
+        } else {
+            builder.setEndpointConfiguration(new EndpointConfiguration("s3.amazonaws.com", null));
```

Comment:
define at different place is better.

Reply:
What place will you suggest? If it is AbstractS3FileInputPlugin, it could not be as this is not applicable for Riak CS.

Reply:
`private final String something = "s3.amazonaws.com"` ?

Reply:
I gotta insist a literal value there is more readable than an extracted name, especially with the context of the (updated) comment above it and when it is referred only once. What do you think?